typings/styled-components.d.ts

```diff
@@ -1,163 +1,176 @@
 import * as React from "react";
-import { StatelessComponent } from "react";
+import { StatelessComponent, ComponentClass } from "react";
 
 interface ThemeProps {
-  theme: Object;
+  theme: any;
 }
 
-type ConstrainedProps<C, P> = C & ({ defaultProps?: P } | { new(props?: P, context?: any): any });
+type Component<P> = ComponentClass<P> | StatelessComponent<P>;
 type StyledProps<P> = P & ThemeProps;
-type Interpolation<P> = ((executionContext: StyledProps<P>) => string) | string | number
+interface InterpolationFunction<P> {
+  (props: StyledProps<P>): InterpolationValue<P> | ReadonlyArray<Interpolation<P>>;
+}
+type InterpolationValue<P> = string | number;
+type Interpolation<P> = InterpolationFunction<P> | InterpolationValue<P> | ReadonlyArray<InterpolationValue<P> | InterpolationFunction<P>>;
+
+type OuterStyledProps<P> = P & {
+  theme?: Object;
+  innerRef?: (instance: any) => void;
```

Comment:
This is a ref function: https://github.com/styled-components/styled-components/blob/master/docs/tips-and-tricks.md#refs-to-dom-nodes